### PR TITLE
Create RELEASE.md

### LINKED ISSUE #3036: Road to Stability
We currently to nightly "unstable" releases and biweekly "alpha" releases, however, at this early stage of the project, we do not have any strategies towards stabilizing the code and the releases.

I use this meta ticket to outline some strategies to revisit in future in case we want to move into a "beta" or "stable" phase.

Useful resouce: 
- https://hackmd.io/HOaKGHWGT-aLzF6kh1KQfA
- I had this call with the Teku, Lighthouse, and Prysm teams last year
- Everything in this document also applies to Lodestar obviously

The key element is the release checklist (we can already tick two boxes):
> Release components checklist for teams:
> - [x] dedicate a team member to commit to releases (release manager, release coordinator) --> @q9f?
> - [x] commit to a strict versioning scheme (semver, calver, etc.) --> **semver** (lerna)
> - [x] enable concurrent development that reflect different expectations of release quality (stable, beta, nightly, etc.)
> - [x] conduct pre-release testing (release candidates, CI, QA)
> - [x] distribute packages (binaries, signatures, checksums) --> **NPM/Docker**

The hackmd above also outlines two different release strategies (Flow vs. Maturity). I always give the example that Geth uses the _"Flow"_ strategy and Parity ~~uses~~ used _"Maturity."_ I'm personally a big fan of the latter.

Current State of Lodestar https://github.com/ChainSafe/lodestar/releases
- [x] has development branches
    - [x] `master` (pull requests default to `master`)
- [x] has releases (23)
    - [x] `v0.29.1` on :warning: `master`
    - [x] `v0.29.0` on :warning: `master`
    - [x] ...
- [x] :warning: has release branches
    - [x] :warning: no release branch for current track (0.29 ?)

### PR DESCRIPTION
**Pull request by philknows**
Persisting release information into the main directory for reference. To help close #3036.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    - Examples: Memory leaks, abnormally high memory usage, abnormally high CPU performance, etc. 
+
+- Has profitibility been affected?
+    - Has profitability decreased and whether or not the variance is acceptable.
+
+- Has any performance metric degraded comparably to the previous `stable` release? If yes, is the variance acceptable?
+    - Example: Block processing times, validator miss ratios, etc.
+
+### Bug fixes to Pre-Releases
+We commit bug fixes exclusively to the specific pre-release candidate and preserve separation from `master` such that we can keep merging unstable PRs to`master`. If a fix is committed, we publish and promote the candidate to `beta.x+1`.
```
Comment (dadepo) [inline RELEASE.md]: Suggestion: Perhaps make it explicit the process (how and when) of having bug fixes committed on pre-release candidate branch back to master?

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Stable Release
+When pre-release candidates have met the testing requirements, we execute `release.yml`.
```
Comment (wemeetagain) [inline RELEASE.md]: include something about tagging the version, eg: `v0.33.0`

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
```
Comment (wemeetagain) [inline RELEASE.md:35]: We should include a section on minor/patch releases, eg: `v0.33.1`

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# Lodestar Release Guidelines
+
+![lodestar-release](https://user-images.githubusercontent.com/58080811/148576989-0e6924f5-8b5a-48c8-baa5-8a833cc364b0.png)
```
Comment (dapplion) [inline RELEASE.md]: This image must be persisted in a more stable place, like committed to the code at `docs/images`

Comment (philknows) [reply RELEASE.md]: Uploaded to docs/images and linked in new RELEASE.md commit 3cd8fcf

File: RELEASE.md (hunk 3 above)
Comment (philknows) [reply RELEASE.md:35]: Added this. Going off of what semver 2.0.0 describes as minor/patch releases.

File: RELEASE.md (hunk 2 above)
Comment (philknows) [reply RELEASE.md]: Added in 3cd8fcf

File: RELEASE.md (hunk 1 above)
Comment (philknows) [reply RELEASE.md]: Added some updates to this in 3cd8fcf. Any feedback on this explicitprocess or did I get it wrong from our conversations?

File: RELEASE.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# Lodestar Release Guidelines
+
+![lodestar-release](https://github.com/ChainSafe/lodestar/blob/master/docs/images/lodestar-releases.png)
```
Comment (dapplion) [inline RELEASE.md]: Links should be relative fs links
```suggestion
![lodestar-release](docs/images/lodestar-releases.png)
```

**Review by dapplion — DISMISSED** [review]

**Review by g11tech — DISMISSED** [review]

File: RELEASE.md (hunk 3 above)
Comment (wemeetagain) [reply RELEASE.md:35]: I was thinking more about when we decide to release them. I think its also related to the maintenance window for a release.

In the past, we've cut a patch release when our previous release had a glaring bug in it.
And we never went back to prior releases and patched them. The current release was the only maintained release that got patches.

Comment (dapplion) [reply RELEASE.md:35]: Agree with @wemeetagain above point

File: RELEASE.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# Lodestar Release Guidelines
+
+![lodestar-release](docs/images/lodestar-releases.png)
+
+## Release Process Rules
+
+1. The team selects a commit from `master` as a "stable target" for a new version release.
+2. The selected commit is branched into a separate release branch (Example: `v0.33.x`) 
+3. The commit on the separate release branch is tagged with `beta.x`. (Example: `v0.33.0-beta.0`) and published for testing as a pre-release.
+4. The team bumps `master` to the next version and continues releasing nightly builds. (Example: `v0.34.0`)
```
Comment (dadepo) [inline RELEASE.md]: Suggestion: maybe make it explicit this is about bumping the version as specified in `package.json`. And nothing to do with branching or tagging etc.

File: RELEASE.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+## Pre-Releases
+A `beta.x` release gets published and tagged when the team selects a commit as a "stable target" candidate.
```
Comment (dadepo) [inline RELEASE.md]: suggestion: Maybe change  occurrences of `beta.x` to something v<y>.beta.x to make it more in line with the proposed format.

File: RELEASE.md (hunk 6 above)
Comment (philknows) [reply RELEASE.md]: Made it more explicit in the new commit.

File: RELEASE.md (hunk 7 above)
Comment (philknows) [reply RELEASE.md]: Standardized, thanks!

File: RELEASE.md (hunk 3 above)
Comment (philknows) [reply RELEASE.md:35]: For patch releases to fix a published stable version, I think it's important to get those fixes out there ASAP and to patch previous versions as well. These would generally be for `high` and `critical` types of bugs you're referring to? Do we want to add any specific parameters to include with it?

**Review by dapplion — APPROVED** [review]

**Pull request merged**